src/libstd/path.rs

```diff
@@ -1893,6 +1909,33 @@ impl<'a> IntoIterator for &'a Path {
     fn into_iter(self) -> Iter<'a> { self.iter() }
 }
 
+macro_rules! impl_eq {
+    ($lhs:ty, $rhs: ty) => {
+        #[stable(feature = "partialeq_path", since = "1.6.0")]
+        impl<'a, 'b> PartialEq<$rhs> for $lhs {
+            #[inline]
+            fn eq(&self, other: &$rhs) -> bool { self.components().eq(other.components()) }
```

Comment:
Could this delegate to the `<Path as PartialEq>::eq` implementation instead of re-encoding the same one here? It's also probably fine to omit `ne` for now as it's not specialized in the main impl either.